cloudify_cli/commands/agents.py

```
@@ -0,0 +1,139 @@
+
+        def worker(dep_id):
+            try:
+                execution = client.executions.start(
+                    dep_id,
+                    workflow_id,
+                )
+
+                execution = wait_for_execution(
+                    client,
+                    execution,
+                    events_handler=threadsafe_events_logger,
+                    include_logs=include_logs,
+                    timeout=900
+                )
+
+                if execution.error:
+                    log_to_summary("Execution of workflow '{0}' for "
+                                   "deployment '{1}' failed. [error={2}]"
+                                   .format(workflow_id,
+                                           dep_id,
+                                           execution.error))
+                else:
+                    threadsafe_log("Finished executing workflow "
+                                   "'{0}' on deployment"
+                                   " '{1}'".format(workflow_id, dep_id))
+
+            except ExecutionTimeoutError as e:
+                log_to_summary("Execution of workflow '{0}' "
+                               "for deployment '{1}' timed out. "
+                               "* Run 'cfy executions cancel "
+                               "--execution-id {2}' to cancel"
+                               " the running workflow."
+                               .format(
+                                    workflow_id,
+                                    deployment_id,
+                                    e.execution_id
+                               ))
+
+        threads = [threading.Thread(target=worker, args=(dep.id,))
+                   for dep in deps]
+
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+        if error_summary:
+            logger.info('Summary:\n{0}\n'.format(
+                '\n'.join(error_summary)
```

Maybe throw an exception when error_summary is not empty?